project.clj

```diff
@@ -1,3 +1,58 @@
+(require 'clojure.java.io)
+
+(import '[java.util.zip ZipInputStream]
+        '[java.io FileOutputStream])
+
+(defmacro while-let [[sym expr] & body]
+  `(loop [~sym ~expr]
+     (when ~sym
+       ~@body
+       (recur ~expr))))
+
+(defn jdk-find [f]
+  (let [home (clojure.java.io/file (System/getProperty "java.home"))
+        parent (.getParentFile home)
+        paths [(clojure.java.io/file home f)
+               (clojure.java.io/file home "lib" f)
+               (clojure.java.io/file parent f)
+               (clojure.java.io/file parent "lib" f)]]
+    (->> paths (filter #(.canRead ^java.io.File %)) first str)))
+
+(def jdk-sources
```

Comment:
These top-level defs can be considered an "inline Lein plugin" for fetching the JDK sources and adding them to the classpath without altering classloaders.

They are not essential to this PR, but they allow the tests depending on the JDK sources being reachable to still pass.

...One could opt instead for disabling the tests when dynapath is disabled, but disabling tests seems delicate (less coverage, less confidence).

It also seems a useful point to show that consumers can add JDK sources _now_, with little code.

Admittedly it would be more tidy to extract this to a Lein plugin, but first I'd like to know what you think of the approach to begin with.

Reply:
It seems reasonable overall. The code itself is faily simple.